File: test/20230311/l.py
n = int(input())
a = []
mod = 998244353
for _ in range(n):
    a.append(tuple(map(int, input().split())))
dp = [[0 for i in range(2)] for j in range(n)]
dp[0][0] = dp[0][1] = 1
for i in range(1, n):
    for j in range(2):
        if a[i-1][0] != a[i][j]:
            dp[i][j] += dp[i-1][0]
        if a[i-1][1] != a[i][j]:
            dp[i][j] += dp[i-1][1]
    dp[i][0] %= mod
    dp[i][1] %= mod
print((dp[n-1][0] + dp[n-1][1]) % mod)

File: nowcoder/2023牛客暑假多校/2023牛客暑假多校第四场/l.cpp
#include <iostream>
#include <cstring>
#include <algorithm>
#include <set>
#include <bitset>
#include <queue>
#include <map>
#include <stack>
#include <random>
#include <cassert>
#include <functional>
#include <iomanip>
#include <array>
#include <unordered_map>
#include <unordered_set>
#define inf 0x3f3f3f3f
#define linf 0x3f3f3f3f3f3f3f3fll
#define endl '\n'
#define int long long
#define ull unsigned long long
#define SZ(x) (int)x.size()
#define rep(i, a, n) for (int i = (a); i <= (n); i++)
#define dec(i, n, a) for (int i = (n); i >= (a); i--)
using namespace std;
using pii = pair<int, int>;
using sis = tuple<string, int, string>;
// mt19937 mrand(random_device{}());
// int rnd(int x) { return mrand() % x; }
const int N = 10 + 1e5, mod = 1e9 + 7;
void solve()
{
    int n, m, q; cin >> n >> m >> q;
    vector<sis> vec(q);
    for(auto &[s, x, y] : vec) cin >> s >> x >> y;
    reverse(vec.begin(), vec.end());
    int sm = 0;
    set<int> str, stc;
    for(auto &[s, x, y] : vec)
    {
        if(s == "row")
        {
            if(str.count(x)) continue;
            n --;
            if(y == "on") sm += m;
            str.insert(x);
        }
        else
        {
            if(stc.count(x)) continue;
            m --;
            if(y == "on") sm += n;
            stc.insert(x);
        }
    }
    cout << sm << endl;
}
signed main()
{
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);
    // int t; cin >> t;
    // while(t--)
        solve();

    return 0;
}